### PR TITLE
solution

### PR DESCRIPTION
**Pull request by ihorandrianov**
[DEMO LINK](https://ihorandrianov.github.io/react_list-of-goods-hooks/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -15,47 +16,98 @@ export const goodsFromServer = [
...
+  return visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortTypes, setSortTypes] = useState<SortType>(SortType.NONE);
+  const [isReverse, setIsReverse] = useState<boolean>(false);
+  const resetState = () => {
+    setSortTypes(SortType.NONE);
+    setIsReverse(false);
+  };
```
Comment (silchyn) [inline src/App.tsx:59]:
```suggestion
  const [sortTypes, setSortTypes] = useState<SortType>(SortType.NONE);
  const [isReverse, setIsReverse] = useState<boolean>(false);
  
  const resetState = () => {
    setSortTypes(SortType.NONE);
    setIsReverse(false);
  };
```

File: src/App.tsx [hunk 2]
```diff
@@ -15,47 +16,98 @@ export const goodsFromServer = [
...
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
+          {getReorderedGoods(goodsFromServer,
+            sortTypes,
+            isReverse).map((good) => {
+            return (
+              <li key={good} data-cy="Good">{good}</li>
+            );
+          })}
```
Comment (silchyn) [inline src/App.tsx:110]:
```suggestion
          {getReorderedGoods(
            goodsFromServer,
            sortTypes,
            isReverse,
           ).map((good) => (
              <li key={good} data-cy="Good">{good}</li>
           ))}
```

**Review by silchyn — CHANGES_REQUESTED** [review]

File: src/App.tsx [hunk 3]
```diff
@@ -15,47 +16,98 @@ export const goodsFromServer = [
...
+    case SortType.LENGTH:
+      visibleGoods.sort((curr, next) => curr.length - next.length);
+      break;
+    case SortType.ALPABET:
+      visibleGoods.sort((curr, next) => curr.localeCompare(next));
+      break;
+    case SortType.NONE:
+      break;
+    default:
+      break;
```
Comment (silchyn) [inline src/App.tsx:43]:
```suggestion
    default:
      break;
```

**Review by silchyn — CHANGES_REQUESTED** [review]

File: src/App.tsx [hunk 4]
```diff
@@ -15,47 +16,97 @@ export const goodsFromServer = [
...
+  ALPABET,
+  LENGTH,
+}
+
+export function getReorderedGoods(
+  goods: string[],
+  sortType: SortType,
+  isReversed: boolean,
+) {
+  // To avoid the original array mutation
```
Comment (naziamoff) [inline src/App.tsx:30]: no need to add comments to things like that, this decision is quite understandable without comment

File: src/App.tsx [hunk 5]
```diff
@@ -15,47 +16,97 @@ export const goodsFromServer = [
...
           Reverse
         </button>
 
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
+        {(isReverse || sortTypes !== SortType.NONE) && (
```
Comment (naziamoff) [inline src/App.tsx:90]: would be good to create a separate variable for that

File: src/App.tsx [hunk 6]
```diff
@@ -15,47 +16,97 @@ export const goodsFromServer = [
...
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
+          {getReorderedGoods(
+            goodsFromServer,
+            sortTypes,
+            isReverse,
+          ).map((good) => (
```
Comment (naziamoff) [inline src/App.tsx:107]: looks totally unclear now. Create separate variable for getReorderedGoods() result and use it here

File: src/App.tsx [hunk 7]
```diff
@@ -15,47 +16,97 @@ export const goodsFromServer = [
...
+            className="button is-danger is-light"
+            onClick={resetState}
+          >
+            Reset
+          </button>
+        )}
       </div>
 
       <ul>
         <ul>
```
Comment (naziamoff) [inline src/App.tsx:102]: do we need nested lists here?

File: src/App.tsx [hunk 8]
```diff
@@ -15,47 +16,97 @@ export const goodsFromServer = [
...
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
+          {getReorderedGoods(
+            goodsFromServer,
+            sortTypes,
+            isReverse,
+          ).map((good) => (
+            <li key={good} data-cy="Good">{good}</li>
```
Comment (naziamoff) [inline src/App.tsx:108]:
```suggestion
            <li key={good} data-cy="Good">
              {good}
            </li>
```

**Review by naziamoff — CHANGES_REQUESTED** [review]

File: src/App.tsx [hunk 9]
```diff
@@ -15,48 +16,99 @@ export const goodsFromServer = [
...
           Reverse
         </button>
 
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
+        {(isReverse || sortTypes !== SortType.NONE) && (
```
Comment (Anna-Tkachenko) [inline src/App.tsx:95]: you didn't fix previous comment
let's create separate variable with understandable name, where you can store the result for this check
and let's use this variable here

**Review by Anna-Tkachenko — APPROVED** [review]

**Pull request closed without merging**